bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -338,6 +338,22 @@ class PlotCanvasView extends Renderer.View
             if is_panning?
               range_info['start'] = rng.get('start')
 
+    min_interval = rng.get('min_interval')
+    max_interval = rng.get('max_interval')
+    if min_interval? || max_interval?
+      new_interval = Math.abs(range_info['end'] - range_info['start'])
+      center = 0.5 * (rng.get('end') + rng.get('start'))
+      sign = if reversed then -1 else 1
+
+      if min_interval > 0
+        if new_interval < min_interval
+          range_info['start'] = center - sign * 0.5 * min_interval
+          range_info['end'] = center + sign * 0.5 * min_interval
+      if max_interval > 0
+        if new_interval > max_interval
+          range_info['start'] = center - sign * 0.5 * max_interval
+          range_info['end'] = center + sign * 0.5 * max_interval
```

Comment:
Setting to a smaller interval around the current center is not the correct behavior in general.  Consider a scroll wheel zoom with the mouse at the far right edge of the plot. The range should shrink largely from the left side only, keeping the right range point exactly as it is. That's not what this code would do tho, it would "jump" do that the current center is still the center and the far right edge (currently under the mouse) would suddenly be shifted out of the plot. 

> On May 25, 2016, at 15:36, Almar Klein <notifications@github.com> wrote:
> 
> In bokehjs/src/coffee/models/plots/plot_canvas.coffee:
> 
> > @@ -338,6 +338,22 @@ class PlotCanvasView extends Renderer.View
> >              if is_panning?
> >                range_info['start'] = rng.get('start')
> >  
> > +    min_interval = rng.get('min_interval')
> > +    max_interval = rng.get('max_interval')
> > +    if min_interval? || max_interval?
> > +      new_interval = Math.abs(range_info['end'] - range_info['start'])
> > +      center = 0.5 * (rng.get('end') + rng.get('start'))
> > +      sign = if reversed then -1 else 1
> > +
> > +      if min_interval > 0
> > +        if new_interval < min_interval
> > +          range_info['start'] = center - sign * 0.5 * min_interval
> > +          range_info['end'] = center + sign * 0.5 * min_interval
> > +      if max_interval > 0
> > +        if new_interval > max_interval
> > +          range_info['start'] = center - sign * 0.5 * max_interval
> > +          range_info['end'] = center + sign * 0.5 * max_interval
> @bryevdv I updated the variable name to new_range. Actually, when the interval is too big/small, it pins the range relative to the current center; there wont be any shifting.
> 
> I don't think canceling the event is a good idea: imagine using the box zoom tool and zooming in too far: nothing would happen. Also scroll wheel is in steps so you would not reach the real minimum/maximum.
> 
> One thing that I am not sure about is that the constraint is per axis, so zooming can still take place in the other dimension unless it is constrained too. This is IMO desired behavior for the box zoom, but perhaps not for scroll zoom. Not sure what to do though. Triaging between these two cases seems awkward. Maybe this is acceptable as users would typically constrain both axis?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>